File: Models/Appointment.js
// models/Appointment.js
const mongoose = require('mongoose');

const appointmentSchema = new mongoose.Schema({
    patient: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
    doctor: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
    appointment_date: { type: Date, required: true },
    purpose: { type: String, required: true },
    status: { type: String, enum: ['scheduled', 'canceled', 'completed'], default: 'scheduled' }
});

appointmentSchema.index({ patient: 1, appointment_date: 1 }); // For efficient queries

module.exports = mongoose.model('Appointment', appointmentSchema);

File: frontend/lib/models/appointment.dart
// lib/models/appointment.dart
class Appointment {
  final String id;
  String patientId;
  String doctorId;
  DateTime start;
  DateTime end;
  String status;
  String purpose;
  bool suspended;

  Appointment({
    required this.id,
    required this.patientId,
    required this.doctorId,
    required this.start,
    required this.end,
    required this.status,
    required this.purpose,
    this.suspended = false, required DateTime dateTime,
  });
}
